go/tasks/plugins/array/k8s/transformer.go

```diff
@@ -81,6 +84,9 @@ func FlyteArrayJobToK8sPodTemplate(ctx context.Context, tCtx core.TaskExecutionC
 		return v1.Pod{}, nil, err
 	}
 
+	annotations := utils.UnionMaps(config.GetK8sPluginConfig().DefaultAnnotations, tCtx.TaskExecutionMetadata().GetAnnotations())
```

Comment:
So `tCtx.TaskExecutionMetadata().GetAnnotations()` is always going to be empty by design?

Reply:
tCtx.TaskExecutionMetadata().GetAnnotations() comes from the workflow/task. On top we add few default ones that are set in Propeller config.

Reply:
yes, currently we miss the default labels/annotations  so adding those in this PR.